src/libPMacc/include/particles/memory/boxes/ParticlesBox.hpp

```diff
@@ -250,49 +230,36 @@ class ParticlesBox
     HDINLINE bool removeLastFrame(const DataSpace<DIM> &idx)
     {
         //!\todo this is not thread save
-        vint_t *lastFrameIdx = &(superCells(idx).LastFrameIdx());
-        const vint_t last_id = *lastFrameIdx;
-
-        const vint_t prev_id = prev[last_id];
-        prev[last_id] = INV_IDX; //delete previous frame of the frame which we remove
+        FrameType** lastFrameNativPtr = &(getSuperCell(idx).lastFramePtr);
 
-        if (prev_id != INV_IDX)
+        FramePtr last(*lastFrameNativPtr);
+        if (last.isValid())
         {
-            //prev_id is Valid
-            next[prev_id] = INV_IDX; //clear next of previous frame
-            *lastFrameIdx = prev_id; //set new last particle
-            removeFrame(data[last_id]);
-            return true;
+
+            FramePtr prev(last->previousFrame);
+            //            last->previousFrame = FramePtr(); //delete previous frame of the frame which we remove
```

Comment:
do you **delete** the previous frame here?
Or do you just set the pointer to the previous frame?

In this case:
after deleting frame, set pointer to previous frame